app/models/dialog_tab.rb

```diff
@@ -26,4 +26,15 @@ def dialog_fields
   def dialog_resources
     dialog_groups
   end
+
+  def validate
+    result = []
+    return result << "Tab #{label} must have at least one Box." if dialog_groups.blank?
+
+    dialog_groups.each do |dg|
+      err_msg = dg.validate
+      result << err_msg unless err_msg.blank?
+    end
+    result
+  end
```

Comment:
This could be more concisely written as:

```ruby
def validate
  return ["Tab #{label} must have at least one Box."] if dialog_groups.blank?

  dialog_groups.collect do |dg|
    err_msg = dg.validate
    err_msg.blank? ? nil : err_msg
  end.compact
end
```